internal/types/document_validation.go

```diff
@@ -79,5 +79,14 @@ func (d *Document) ValidateData() error {
 		return newValidationError(fmt.Errorf("invalid document: document must contain '_id' field"))
 	}
 
+	v := must.NotFail(d.Get("_id"))
+
+	switch v.(type) {
+	case Array, *Array:
```

Comment:
```suggestion
	case *Array:
```

Array should never be passed by value